### PR TITLE
Unify asprintf/vasprintf implementations and make them truly portable by using vsnprintf

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]
Nice cleanup!

File: include/swift/Runtime/Portability.h [hunk 1]
```diff
@@ -35,11 +35,12 @@ inline static int
 swift_asprintf(char **strp, const char *fmt, ...) {
   va_list args;
   va_start(args, fmt);
-#if defined(_WIN32)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wuninitialized"
-  int len = _vscprintf(fmt, args);
-#pragma GCC diagnostic pop
+
```
Comment (mikeash) [inline include/swift/Runtime/Portability.h:38]: Can we call through to `swift_vasprintf` and remove the duplicate implementation?

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by mikeash — APPROVED** [review]
Hooray!

File: include/swift/Runtime/Portability.h (hunk 1 above)
Comment (kubamracek) [reply include/swift/Runtime/Portability.h]: We can and we should! 👍

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Pull request merged**